### PR TITLE
adding filters parameter into stock inventories report

### PR DESCRIPTION
**Pull request by jeremielodi**
This PR helps to add filter parameters into the stock inventories report, in order to discribe the result

### CODE REVIEW

File: server/controllers/stock/reports/index.js
Comment (jniles) [inline server/controllers/stock/reports/index.js]: Hmm.... this isn't really extensible.  Also, this method doesn't allow keys to be translated.

Why don't you try using something like like the `formatFilters()` function from the patient registration report?  It lists every key and it's translation value ... like this:
```js
// translation key mappings for dynamic filters
// Basically, to show a pretty filter bar, this will translate URL query params
// into human-readable text to be placed in the report, showing the properties
// filtered on.
function formatFilters(qs) {
  const columns = [
    { field : 'name', displayName : 'FORM.LABELS.NAME' },
    { field : 'sex', displayName : 'FORM.LABELS.GENDER' },
    { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
    { field : 'reference', displayName : 'FORM.LABELS.REFERENCE' },
    { field : 'dateBirthFrom', displayName : 'FORM.LABELS.DOB', comparitor : '>', isDate : true },
    { field : 'dateBirthTo', displayName : 'FORM.LABELS.DOB', comparitor : '<', isDate : true },
    { field : 'dateRegistrationFrom', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '>', isDate : true },
    { field : 'dateRegistrationTo', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '<', isDate : true },
    { field : 'debtor_group_uuid', displayName : 'FORM.LABELS.DEBTOR_GROUP' },
    { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
    { field : 'user_id', displayName : 'FORM.LABELS.USER' },
    { field : 'limit', displayName : 'FORM.LABELS.LIMIT' },
    { field : 'period', displayName : 'TABLE.COLUMNS.PERIOD', isPeriod : true },
  ];

  return columns.filter(column => {

    const value = qs[column.field];

    if (!_.isUndefined(value)) {
  
      if (column.isPeriod) {
        const service = new PeriodService(new Date());
        column.value = service.periods[value].translateKey;
      } else {
        column.value = value;
      }
      return true;
    }
    return false;
  });
} 
```

This format has the following advantages:
 1. It's clear which filters are supported and which are not.
 2. The display name can be anything found in the translation dictionary.
 3. No need to `try/catch` since the code will ignore anything that isn't in the supported filters object.  
 4. Returns an array as required.
 5. Allows any additional properties to be added to a given filter (like the `isDate` property or the `comparitor` property).

File: server/controllers/stock/reports/index.js
Comment (jniles) [inline server/controllers/stock/reports/index.js]: Errors should almost always be handled in some way in production.  There has to be a really good reason to have a blank `catch`.  At minimum, we should log something using `winston.error()` to let the application developers know that something when wrong.

**Comment by jniles** [comment]
@jeremielodi, let me know when you need a second review.

**Comment by jeremielodi** [comment]
Hi! Try just to run branch and see the result, if it good I'll just remove
unimportant codes in handlebar files and push the last changes

**Comment by jniles** [comment]
@jeremielodi, please push the latest changes for me to analyse.  It doesn't do any good for me to look at old material.  If those changes are not ready, we can wait on the review until they are ready.

File: server/controllers/stock/reports/index.js
Comment (jniles) [inline server/controllers/stock/reports/index.js]: This code is no longer necessary.  Please remove it.

File: server/controllers/stock/reports/index.js
Comment (jniles) [inline server/controllers/stock/reports/index.js]: This code is no longer necessary.  Please remove it.

File: server/controllers/stock/reports/index.js
Comment (jniles) [inline server/controllers/stock/reports/index.js]: Do you need this array?  It looks like it is never used.

**Review by jniles — CHANGES_REQUESTED** [review]
It looks like there is still some legacy code in here that should be removed.  After that, LGTM.

**Comment by jeremielodi** [comment]
closes #1721

**Pull request merged**
